### PR TITLE
Etcd

### PR DESCRIPTION
**Pull request by mceloud**

### CODE REVIEW

**Comment by fpytloun** [comment]
Please rebase

**Comment by mceloud** [comment]
done

File: kubernetes/master/kubelet.sls [hunk 1]
```diff
@@ -12,6 +12,14 @@
 /etc/kubernetes/config:
   file.absent
 
+{%- for namespace in master.namespace %}
```
Comment (fpytloun) [inline kubernetes/master/kubelet.sls]: Pillar structure should be namespace*s* when it's list and not dictionary.

File: kubernetes/files/manifest/kube-controller-manager.manifest [hunk 2]
```diff
@@ -16,7 +16,7 @@ spec:
     - /bin/sh
     - -c
     - /usr/local/bin/kube-controller-manager
-      --master=127.0.0.1:8080
+      --master={{ master.apiserver.insecure_address }}:8080
       --cluster-name=kubernetes
```
Comment (fpytloun) [inline kubernetes/files/manifest/kube-controller-manager.manifest]: Don't you want to have cluster name configurable?

File: kubernetes/files/manifest/kube-controller-manager.manifest [hunk 3]
```diff
@@ -16,7 +16,7 @@ spec:
...
       --service-account-private-key-file=/etc/ssl/private/kubernetes-server.key
```
Comment (fpytloun) [inline kubernetes/files/manifest/kube-controller-manager.manifest]: How is that ssl key deployment handled? It's automatically generated after package installation or how?

File: kubernetes/files/manifest/kube-controller-manager.manifest (hunk 2 above)
Comment (mceloud) [inline kubernetes/files/manifest/kube-controller-manager.manifest]: Probably yes, but not part of this pull request.

File: kubernetes/files/manifest/kube-controller-manager.manifest (hunk 3 above)
Comment (mceloud) [inline kubernetes/files/manifest/kube-controller-manager.manifest]: SSL key deployment is done by salt-formula-salt.

File: kubernetes/master/kubelet.sls (hunk 1 above)
Comment (mceloud) [inline kubernetes/master/kubelet.sls]: done
